File: C2/PDOO/Practicas/DeepSpaceRuby/lib/PowerEfficientSpaceStation.rb

#@author Pedro Pablo Ruiz Huertas y Juan Antonio Villegas Recio

#encoding:utf-8
require_relative "Transformation"
require_relative "SpaceStation"


module Deepspace
  class PowerEfficientSpaceStation < SpaceStation
    
    @@EFFICIENCYFACTOR=1.1
    
    def initialize(station)
      createStation(station)
    end
    
    def fire
      factor=super*@@EFFICIENCYFACTOR
      return factor
    end
    
    def protection
      factor=super*@@EFFICIENCYFACTOR
      return factor
    end
    
    def setLoot(loot)
      super
      if loot.efficient
        return Transformation::GETEFFICIENT
      else
        return Transformation::NOTRANSFORM
      end
    end
    
    def to_s
      return "POWER EFFICIENT SPACE STATION"+super
    end
    
  end
end

File: C2/PDOO/Practicas/DeepSpaceJava/src/deepspace/PowerEfficientSpaceStation.java


package deepspace;

/**
 *
 * @author Pedro Pablo Ruiz Huertas y Juan Antonio Villegas Recio
 */
public class PowerEfficientSpaceStation extends SpaceStation{
    
    private static float EFFICIENCYFACTOR=1.10f;
    
    PowerEfficientSpaceStation(SpaceStation station){
        super(station);
    }
    @Override
    public float fire(){
        return super.fire()*EFFICIENCYFACTOR;
    }
    
    @Override
    public float protection(){
        return super.protection()*EFFICIENCYFACTOR;
    }
    
    @Override
    public Transformation setLoot(Loot loot){
        super.setLoot(loot);
        if (loot.getEfficient())
            return Transformation.GETEFFICIENT;
        else
            return Transformation.NOTRANSFORM;
    }

    @Override
    public String toString() {
        return "POWER EFFICIENT SPACE STATION:\n"+super.toString();
    }
}
